### ISSUE TITLE
Support Jobs API

### ISSUE THREAD

**Issue by astronouth7303** [python-gitlab/python-gitlab#262]
Currently, the [Jobs API](https://docs.gitlab.com/ce/api/jobs.html) appears to be completely unsupported. This is important if you want to do anything with, say, artifacts.

**Comment by gpocentek** [comment]
The build API (``ProjectBuild``) seems to do the same things but is probably deprecated or removed upstream.

**Comment by astronouth7303** [comment]
Not removed or deprecated, but is [meant for runners only](https://docs.gitlab.com/ce/api/ci/builds.html).

**Comment by astronouth7303** [comment]
The `ProjectBuild.artifacts()` method is returning a 404 for me against GitLab 9.2.1.

**Comment by gpocentek** [comment]
Jobs API support has been added in python-gitlab 0.21.1, but you have to use the v4 API (see "GitLab v4 support" in http://python-gitlab.readthedocs.io/en/stable/api-usage.html#gitlab-gitlab-class):
`````python
gl = gitlab.Gitlab(..., api_version=4)
project = gl.projects.get(id)
print(project.jobs.list())
`````

For v3 you have to use the ``builds`` manager (https://github.com/gitlabhq/gitlabhq/blob/8-16-stable/doc/api/builds.md):
`````python
gl = gitlab.Gitlab(..., api_version=3)
project = gl.projects.get(id)
print(project.builds.list())
`````

This build API is not the same as the CI build API (not implemented in python-gitlab).

**Comment by astronouth7303** [comment]
`ProjectJob.artifacts()` raises a 404 for me. Is this API an EE-only feature?

**Comment by gpocentek** [comment]
This is supported in the community edition https://docs.gitlab.com/ce/api/jobs.html#get-job-artifacts. job.artifacts() should return something according to the docs, but it's not. I need to dig a little more to understand what's happening.

**Comment by astronouth7303** [comment]
:woman_facepalming:

The API is documented to return a 404 when the job is valid, but it produced no artifacts. So probably the only bug is that the API raises an error instead of returning `None` (or whatever).

**Comment by novalis** [comment]
It looks like projects don't have a jobs_enabled field yet.

**Comment by gpocentek** [comment]
@novalis could you give a bit more detail about jobs_enabled? Is it something missing in python-gitlab?

**Comment by novalis** [comment]
Yes, it's a field on [project](https://docs.gitlab.com/ce/api/projects.html).  It replaces builds_enabled.

**Comment by gpocentek** [comment]
Should be fixed in d1e7cc797. Thanks for the warning.
